Sprint 13

brozozowska opened the pull request:
Задания спринта 13 выполнены.

brozozowska committed:
refactor: вынесены лишние ответственности из WebViewViewController в WebViewPresenter

brozozowska committed:
refactor: вынесены лишние ответственности из WebViewPresenter в AuthHelper

brozozowska committed:
test: добавлены юнит-тесты для WebViewViewController

brozozowska committed:
refactor: вынесены лишние ответственности из ProfileViewController в ProfilePresenter
дополнительно добавлена разметка в AuthHelper, WebViewPresenter и WebViewViewController

brozozowska committed:
refactor: вынесены лишние ответственности из ImagesListViewController в ImagesListPresenter

brozozowska committed:
test: добавлены юнит-тесты для ProfileViewController
дополнительно добавлены ProfileServiceProtocol, ProfileImageServiceProtocol и инициализатор профиля

brozozowska committed:
test: добавлены юнит-тесты для ImagesListViewController
дополнительно добавлен ImagesListServiceProtocol

brozozowska committed:
test: добавлены ui-тесты для WebView, ImagesList и ProfileView
дополнительно добавлены accessibilityIdentifier для соответствующих элементов

brozozowska committed:
test: отключена пагинация при проведении ui-тестов

Helvetii951 commented on ImageFeed/ImagesList/ImagesListPresenter.swift:
```diff
@@ -0,0 +1,121 @@
+    private var photos: [Photo] = []
+    private let service: ImagesListServiceProtocol
+    private static let dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.locale = Locale(identifier: "ru_RU")
+        formatter.dateStyle = .long
+        formatter.timeStyle = .none
+        return formatter
```
Лучше не привзваться к конкретной локали. Пусть отображение даты будет отличаться от дизайна, но пользователь увидит дату в привычном для себя виде.

Helvetii951 commented on ImageFeed/ImagesList/ImagesListPresenter.swift:
```diff
@@ -0,0 +1,121 @@
+        let photoID = photo.id
+        let isLike = !photo.isLiked
+        
+        UIBlockingProgressHUD.show()
+        
+        service.changeLike(photoID: photoID, isLike: isLike) { [weak self] result in
+            guard let self else { return }
+            
+            defer { UIBlockingProgressHUD.dismiss() }
```
Лучше не помещать работу с UI(`UIBlockingProgressHUD.show()`) в презентер. Пусть он занимается только бизнес-логикой.

brozozowska replied:
UIBlockingProgressHUD больше не вызывается напрямую из презентера, а перенесён в методы ImagesListViewController.

brozozowska committed:
fix: исправлены замечания код-ревью
отображение даты отвязано от конкретной локали
убрана работа UIBlockingProgressHUD из презентера
в ImagesListViewControllerProtocol добавлены методы по работе с UIBlockingProgressHUD
в ImagesListViewController и ImagesListViewTests добавлены методя для соответствия протоколу
